The actual ABI compatibility of sqlite3 doesn't seem to depend on the
patchlevel (the x in 3.7.x), so omit that in our sanity check here.

This matters in Debian because the sqlite3 library package exports a
slightly more liberal shlibs file than the library version itself.


--- a/subversion/libsvn_subr/sqlite.c
+++ b/subversion/libsvn_subr/sqlite.c
@@ -486,7 +486,7 @@ static volatile svn_atomic_t sqlite_init
 static svn_error_t *
 init_sqlite(void *baton, apr_pool_t *pool)
 {
-  if (sqlite3_libversion_number() < SQLITE_VERSION_NUMBER) {
+  if (sqlite3_libversion_number()/1000 < SQLITE_VERSION_NUMBER/1000) {
     return svn_error_createf(SVN_ERR_SQLITE_ERROR, NULL,
                              _("SQLite compiled for %s, but running with %s"),
                              SQLITE_VERSION, sqlite3_libversion());
